api/v1alpha1/httproute_types.go

```diff
@@ -192,7 +192,52 @@ type HTTPRouteAction struct {
 
 // HTTPRouteStatus defines the observed state of HTTPRoute
 type HTTPRouteStatus struct {
-	Gateways []core.ObjectReference `json:"gateways"`
+	// Gateways is a list of the Gateways that are associated with the
+	// HTTPRoute, and the status of the route with respect to each of these
+	// Gateways.  Each associated Gateway should have exactly one entry in
+	// this list.
+	Gateways []RouteGatewayStatus `json:"gateways"`
```

Comment:
Open question:  What happens if a gateway controller adds a status condition to a route, and the controller is subsequently deleted?  What is responsible for clearing out old status conditions?

Reply:
Some of the existing ingress controllers do have a tunable setting [https://kubernetes.github.io/ingress-nginx/user-guide/cli-arguments/](https://kubernetes.github.io/ingress-nginx/user-guide/cli-arguments/) to control this behavior.
We should either suggest all implementers to take the same route or leave this out for v1alpha1 (by adding an issue to track it) and come back to it later.